cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/DatastoreWrapper.java

```diff
@@ -88,6 +90,25 @@ private String getCheckpointDocumentId(String replicatorIdentifier) {
         return "_local/" + replicatorIdentifier;
     }
 
+    public void bulkInsert(List<BasicPullStrategy.BatchItem> batches, boolean pullAttachmentsInline) throws DocumentException  {
+        List<ForceInsertItem> itemsToInsert = new ArrayList<ForceInsertItem>();
+        for (BasicPullStrategy.BatchItem batch : batches) {
+            for (DocumentRevs documentRevs : batch.revsList) {
+                logger.log(Level.FINEST, "Bulk inserting document revs: %s", documentRevs);
+
+                BasicDocumentRevision doc = DocumentRevsUtils.createDocument(documentRevs);
+
+                List<String> revisions = DocumentRevsUtils.createRevisionIdHistory(documentRevs);
+                Map<String, Object> attachments = documentRevs.getAttachments();
+                itemsToInsert.add(new ForceInsertItem(doc, revisions, attachments, batch.attachments, pullAttachmentsInline));
+
+            }
+        }
+        dbCore.forceInsert(itemsToInsert);
+
+    }
+
+
     public void bulkInsert(DocumentRevsList documentRevsList, Map<String[],List<PreparedAttachment>> preparedAttachments, boolean pullAttachmentsInline) throws DocumentException  {
```

Comment:
I would yes, it's inefficient, we aren't tied to the API yet, the only thing that might be a problem is the usage of @snowch's p2p lib